homeassistant/components/halohome/__init__.py

```diff
@@ -0,0 +1,44 @@
+"""Integrate HALO Home into home assistant."""
+import logging
+
+import aiohttp
+import halohome
+from halohome import Connection, HaloHomeError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_LOCATIONS, DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up HALO Home from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    try:
+        locations = await halohome.list_devices(
+            entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD]
+        )
+    except (HaloHomeError, OSError, aiohttp.ClientError) as exception:
+        _LOGGER.error("Caught exception refreshing HALO devices: %s", exception)
```

Comment:
Raise `ConfigEntryNotReady` and don't continue set up

Reply:
If you can figure out that the auth was unsuccessful, `return False` instead.

Reply:
In this case the intention is the utilize the cached devices which are more than likely still up to date, so the integration is still usable when the outside internet is not available.